README.md

```diff
@@ -134,30 +151,167 @@ or in the case of an error:
 ```ts
 type UpdateProfileResponse = {
   error: string;
-  message: string;
 };
 ```
 
-This route lets someone perform partial updates to their profile. If `name` or
+This route lets the user perform partial updates to their profile. If `name` or
 `nft` is `null`, that field is cleared. If either is `undefined` or omitted,
-nothing happens to that field.
+nothing happens to that field. The name must be unique, at least 1 character
+long, at most 32 characters long, and only contain alphanumeric characters,
+periods, and underscores. The NFT must be owned by any of the profile's public
+keys. If the NFT has no image, it will fail.
+
+When creating a profile for the first time, a set of chain IDs can optionally be
+provided. If provided, the public key will be registered for all of these
+chains. If no chain IDs are provided, the public key will only be registered for
+the chain used to sign the request.
 
 The nonce from the latest GET request must be provided to prevent replay
 attacks. It starts at 0, and the GET request will return an empty profile with a
-nonce of 0 if nothing has been set. The name must be unique, at least 1
-character long, and at most 32 characters long. The NFT must be owned by the
-public key, and the signature must be made by the same public key. If the NFT
-has no image, it will fail.
+nonce of 0 if nothing has been set.
+
+The signature is derived by calling `OfflineAminoSigner`'s `signAmino` function
+with the `signDoc` argument generated using `makeSignDoc` from the
+`@cosmjs/amino` package. This can be seen in the signature verification code
+located in [src/index.ts](./src/utils/auth.ts#L20) around line 20.
+
+### `POST /register`
+
+The expected request body type is:
+
+```ts
+type RegisterPublicKeyRequest = {
+  data: {
+    publicKeys: {
+      data: {
+        allow: string;
+        chainIds?: string[];
+        auth: {
+          type: string;
+          nonce: number;
+          chainId: string;
+          chainFeeDenom: string;
+          chainBech32Prefix: string;
+          publicKey: string;
+        };
+      };
+      signature: string;
+    }[];
+    auth: {
+      type: string;
+      nonce: number;
+      chainId: string;
+      chainFeeDenom: string;
+      chainBech32Prefix: string;
+      publicKey: string;
+    };
+  };
+  signature: string;
+};
+```
+
+The returned type is:
+
+```ts
+type RegisterPublicKeyResponse = {
+  success: true;
+};
+```
+
+or in the case of an error:
+
+```ts
+type RegisterPublicKeyResponse = {
+  error: string;
+};
+```
+
+This route lets the user register public keys for their profile and/or set chain
+preferences for their registered public keys. The keys being registered must
+sign allowances (using the same nonce) to allow the profile to register them (or
```

Comment:
What's an allowance? Is that a message that you sign? Is it a term of art in Cosmos/web3?

Reply:
nope, i made it up